src/borg/cache.py

```diff
@@ -612,15 +613,12 @@ def create_master_idx(chunk_idx):
             master_index_capacity = int(len(self.repository) / ChunkIndex.MAX_LOAD_FACTOR)
             if archive_ids:
                 chunk_idx = None
-                if self.progress:
-                    pi = ProgressIndicatorPercent(total=len(archive_ids), step=0.1,
-                                                  msg='%3.0f%% Syncing chunks cache. Processing archive %s',
-                                                  msgid='cache.sync')
+                pi = ProgressIndicatorPercent(total=len(archive_ids), step=0.1,
+                                              msg='%3.0f%% Syncing chunks cache. Processing archive %s',
+                                              msgid='cache.sync')
                 archive_ids_to_names = get_archive_ids_to_names(archive_ids)
-                for archive_id in archive_ids:
-                    archive_name = archive_ids_to_names.pop(archive_id)
-                    if self.progress:
-                        pi.show(info=[remove_surrogates(archive_name)])
+                for archive_id, archive_name in archive_ids_to_names.items():
```

Comment:
guess this processes archives in random order while it was ordered-by-time before?

could look a bit unpretty and also impact performance because when processed in-order, there is more similarity between archives.